code/modules/religion/sparring/sparring_contract.dm

```diff
@@ -0,0 +1,131 @@
+	///who has signed this contract
+	var/list/signed_by = list(null, null)
+
+/obj/item/sparring_contract/Initialize()
+	. = ..()
+	name = "[GLOB.deity]'s sparring contract"
+
+/obj/item/sparring_contract/Destroy()
+	QDEL_NULL(signed_by)
+	. = ..()
+
+/obj/item/sparring_contract/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SparringContract", name)
+		ui.open()
+
+/obj/item/sparring_contract/ui_static_data(mob/user)
+	var/list/data = list()
+	var/area/arena = GLOB.areas_by_type[arena_condition]
+	data["set_weapon"] = weapons_condition
+	data["set_area"] = arena?.name
+	data["set_stakes"] = stakes_condition
+	data["possible_areas"] = get_possible_areas()
+
+	return data
+
+/obj/item/sparring_contract/ui_data(mob/user)
+	var/list/data = list()
+	var/area/arena = GLOB.areas_by_type[arena_condition]
+	var/mob/living/carbon/human/left_partner = signed_by[1]
+	var/mob/living/carbon/human/right_partner = signed_by[2]
+	data["in_area"] = ((left_partner && right_partner && arena) && (left_partner in arena.contents) && (right_partner in arena.contents))
+	data["no_chaplains"] = (!left_partner?.mind?.holy_role && !right_partner?.mind?.holy_role)
+	data["left_sign"] = left_partner ? left_partner.real_name : "none"
+	data["right_sign"] = right_partner ? right_partner.real_name : "none"
+	return data
+
+/obj/item/sparring_contract/proc/get_possible_areas()
+	var/list/area_names = list()
+	var/datum/religion_sect/spar/sect = GLOB.religious_sect
+	for(var/key in sect.arenas)
+		area_names += key
+	return area_names
+
+/obj/item/sparring_contract/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/user = usr
+
+	if(!ishuman(user))
+		to_chat(user, span_warning("This contract refuses to be signed by a lesser creature such as yourself."))
+		return
+
+	var/datum/religion_sect/spar/sect = GLOB.religious_sect
+
+	var/list/resolved_opponents = list()
+	for(var/datum/weakref/resolve_me as anything in sect.past_opponents)
+		var/resolved = resolve_me.resolve()
+		if(!isnull(resolved))
+			resolved_opponents += resolved
+
+	if(user in resolved_opponents && params["stakes"] == HOLY_MATCH)
+		to_chat(user, span_warning("This contract refuses to be signed up for a holy match by a previous holy match loser. Pick a different stake!"))
+
+	//any updating of the terms should update the UI to display new terms
+	. = TRUE
+
+	switch(action)
+		if("fight")
+			var/mob/living/carbon/human/left_partner = signed_by[1]
+			var/mob/living/carbon/human/right_partner = signed_by[2]
+			if(!left_partner || !right_partner || !left_partner.mind || !right_partner.mind)
+				return
+			if(HAS_TRAIT(left_partner, TRAIT_SPARRING) || HAS_TRAIT(right_partner, TRAIT_SPARRING))
+				to_chat(user, span_warning("One participant is already sparring!"))
+				return
+			var/chaplain = left_partner.mind.holy_role ? left_partner : right_partner
+			var/opponent = left_partner.mind.holy_role ? right_partner : left_partner
```

Comment:
what happens if both participants are holy? latejoin chaplains are a thing, as are syndicate bibles.

Reply:
still works, actually. i just suppose that two chaplains shouldn't holy match.